deps/wazuh_testing/wazuh_testing/tools/performance/csv_parser.py

```diff
@@ -102,13 +121,37 @@ def _calculate_all_df_stats(self, dfs_dict):
 
         for node, files in dfs_dict.items():
             for phase in [self.SETUP_PHASE, self.STABLE_PHASE]:
-                for file_name, file_df in files.items():
+
+                # Check whether we are calculating the resources' stats
+                for file_df in files.values():
+                    all_files = self._add_child_process_to_parent_process(files) if "Daemon" in file_df else files
+                    break
+
+                for file_name, file_df in all_files.items():
```

Comment:
Since `phase` is not needed as a parameter for `_add_child_process_to_parent_process`, it is more efficient to call that function outside said loop. Otherwise, it has to calculate the same information multiple times.

On the other hand, it is a bit strange to make a loop with a single iteration to determine if `Default` is inside the dataframe. I would say it's better to use something like `all(...)` and a loop inside. It will check that all dataframes inside `files` contain `Default`, not just the first one. In addition, if the same variable name is used (`files`) instead of a new one (`all_files`), you can avoid this condition:  `if "Daemon" in file_df else files`

```suggestion
            if all("Daemon" in df for df in files.values()):
                # Ensure these dataframes contain resources' stats before trying to add up all children.
                files = self._add_child_process_to_parent_process(files)
            for phase in [self.SETUP_PHASE, self.STABLE_PHASE]:
                for file_name, file_df in files.items():
```